### ISSUE TITLE
Cannot run program "Handbrake CLI". No such file or directory.

### ISSUE THREAD

**Issue by jlhunter2008** [shannah/handbrake-watcher#3]
I admit I am a noob at this but I created the .config file and entered the path to HandbrakeCLI which is in my Applications folder, but I am still getting this error message. Any help would be appreciated.

**Comment by jlhunter2008** [comment]
Just to better help you under I am attaching the .conf file I created. I am running this
 on a Mac Mini but the folders being watched are on my Synology NAS.

[handbrake.properties.txt](https://github.com/shannah/handbrake-watcher/files/1583317/handbrake.properties.txt)

**Comment by shannah** [comment]
You're missing the `=` sign.

~~~~
handbrakecli /Volumes/Macintosh%20HD/Users/"myusername"/Library/Applications/HandBrakeCLI
~~~~

should be 

~~~~
handbrakecli=/Volumes/Macintosh%20HD/Users/"myusername"/Library/Applications/HandBrakeCLI
~~~~

Also, I'm not sure about that encoded space in the path.  I think you'll just want to leave it as a space, rather than "%20"

**Comment by jlhunter2008** [comment]
Steve,

I am still getting the same "java.io.IOException: Cannot run program "HandBrakeCLI": error=2, No such file or directory” error message.

**Comment by shannah** [comment]
It doesn't seem to be picking up the config file.  There are other ways to make it find HandBrakeCLI though:

1. Just add the directory containing it to your PATH before running the script.

e.g.

~~~~
export PATH=$PATH:"/Path/to/Folder/Containing/HandBrakeCLI"
~~~~

Then run handbrake watcher.

You can ensure that it can find it by using `which`. E.g.

~~~~
which HandBrakeCLI
~~~~

This should show you the full path to your HandBrakeCLI binary.  If it doesn't then adding to your PATH didn't work.

2. You could add it on the command line

~~~~
handbrake-watcher -Dhandbrakecli=/path/to/HandBrakeCLI
~~~~

**Comment by jlhunter2008** [comment]
Should it be a .config or .conf file?

**Comment by shannah** [comment]
It should be called "handbrake.properties", and it should be located in the same directory you are running handbrake-watcher in.

**Comment by jlhunter2008** [comment]
The issue was with the filename of the properties file. Thank you! However, now I get this error message.

Watching .
[15:19:41] hb_init: starting libhb thread
[15:19:41] thread 700008e7f000 started ("libhb")
unknown option (--first-subtitle=)
Failed to convert file
Exit code 1

I am trying to get the first english subtitle, which first-subtitle should do since I put eng for subtitle-lang-list

Any help would be appreciated.

Thanks.

**Comment by shannah** [comment]
Command-line "flags" that don't have an associated value should be entered in the handbrake.flags property.  E.g.

~~~~
handbrake.flags=--first-subtitle --another-flag --a-third-flag --etc
~~~~

**Comment by shannah** [comment]
BTW.  Where did you hear about handbrake-watcher?  I made it a while ago mainly for my own use, and I haven't received any issues (or indication that it was being used by others) until last week.

**Comment by jlhunter2008** [comment]
I found it in Google search and I’ll be honest it’s terrific.

**Comment by sebsez** [comment]
Hi Steve, 

I just stumbled upon this and it's perfect for the automation I want to put in place. 
I'm having an issue getting this working, and admit I know very little about coding so I'm certain it's something fundamental I'm not doing... anyway, I was hoping you could point me in the right direction. 

I'm also getting "Cannot run program "HandBrakeCLI": error=2, No such file or directory", but only when I run handbrake-watcher without the "handbrake.properties" file in the directory I'm trying to run it from. 

With the "handbrake.properties" saved in the directory I just get the following and no action, i.e. I have .mkv files in the directory, but the conversion process does not get initiated... 

"Handbrake Watcher v1.0
Watching ."

I've attached a copy of my "handbrake.properties" file

[handbrake.txt](https://github.com/shannah/handbrake-watcher/files/1592724/handbrake.txt)

Any help would be appreciated. 

Thanks 

Seb

**Comment by shannah** [comment]
@sebsez Probably better to start a new issue rather than include yours as a reply to an existing one.

I'm not sure why it's not doing any conversions.  In looking at the code, it looks like the extension check is case sensitive.  E.g. it will convert files with ".mkv" but not ".MKV".  Is it possible that your files have the latter extension.

This is something that should probably be made case-insensitive.

**Comment by sebsez** [comment]
@shannah - Thanks for your response... I've actually found the issue.... When I originally created the "handbrake.properties" it was using mac text edit, which initially creates a .rtf file.... this had some inherent format properties in it, so when I renamed the file to "handbrake.properties", watcher picked these up and did not like them..... I started again with a plain text file and all came good! 

Thanks again. 

PS: I do have another minor issue, I'll start a new thread for it.
